internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -341,6 +355,32 @@ func resourceOrchestratedVirtualMachineScaleSet() *pluginsdk.Resource {
 					}
 				}
 
+				upgradeMode := virtualmachinescalesets.UpgradeMode(diff.Get("upgrade_mode").(string))
+				rollingUpgradePolicyRaw := diff.Get("rolling_upgrade_policy").([]interface{})
+
+				shouldHaveRollingUpgradePolicy := upgradeMode == virtualmachinescalesets.UpgradeModeAutomatic || upgradeMode == virtualmachinescalesets.UpgradeModeRolling
+				if !shouldHaveRollingUpgradePolicy && len(rollingUpgradePolicyRaw) > 0 {
+					return fmt.Errorf("a `rolling_upgrade_policy` block cannot be specified when `upgrade_mode` is set to %q", string(upgradeMode))
+				}
+				shouldHaveRollingUpgradePolicy = upgradeMode == virtualmachinescalesets.UpgradeModeRolling
+				if shouldHaveRollingUpgradePolicy && len(rollingUpgradePolicyRaw) == 0 {
+					return fmt.Errorf("a `rolling_upgrade_policy` block must be specified when `upgrade_mode` is set to %q", string(upgradeMode))
+				}
+
+				hasHealthExtension := false
+				if v, ok := diff.GetOk("extension"); ok {
+					var err error
+					_, hasHealthExtension, err = expandOrchestratedVirtualMachineScaleSetExtensions(v.(*pluginsdk.Set).List())
+					if err != nil {
+						return err
+					}
+				}
+
+				// Virtual Machine Scale Set with Flexible Orchestration Mode and 'Rolling' upgradeMode must have Health Extension Present
+				if upgradeMode == virtualmachinescalesets.UpgradeModeRolling && !hasHealthExtension {
+					return fmt.Errorf("a health extension must be specified when `upgrade_mode` is set to Rolling")
```

Comment:
Minor suggestion to stay consistent with how we usually format error messages, will require an update to the test on ln313 of `orchestrated_virtual_machine_scale_set_resource_other_test.go` as well

```suggestion
					return fmt.Errorf("a health extension must be specified when `upgrade_mode` is set to `Rolling`")
```

Reply:
updated. The new code is in `resourceOrchestratedVirtualMachineScaleSetCreate` method.